### PR TITLE
Move KitodoServiceLoader to API

### PR DESCRIPTION
**Pull request by Beacze**

### CODE REVIEW

File: Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java [hunk 1]
```diff
@@ -27,14 +27,15 @@
...
 import org.apache.logging.log4j.Logger;
-import org.kitodo.config.ConfigMain;
 
 public class KitodoServiceLoader<T> {
     private Class clazz;
+    private String pluginPath;
     private static final Logger logger = LogManager.getLogger(KitodoServiceLoader.class);
 
-    public KitodoServiceLoader(Class clazz) {
+    public KitodoServiceLoader(Class clazz, String pluginPath) {
```
Comment (henning-gerhardt) [inline Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java]: Please add / adjust the JavaDoc for this constructor method.

File: Kitodo-API/pom.xml [hunk 2]
```diff
@@ -28,4 +28,12 @@
     <artifactId>kitodo-api</artifactId>
     <name>Kitodo - API</name>
 
+    <dependencies>
+        <dependency>
+            <groupId>org.apache.logging.log4j</groupId>
+            <artifactId>log4j-slf4j-impl</artifactId>
```
Comment (henning-gerhardt) [inline Kitodo-API/pom.xml]: You are sure about this artifact id? In the moved file I see no class usage for `slf4j` or did I missed something?

Comment (Beacze) [inline Kitodo-API/pom.xml]: import org.apache.logging.log4j.LogManager;
import org.apache.logging.log4j.Logger;

this two classes has automatically imported this dependancy

File: Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java (hunk 1 above)
Comment (Beacze) [inline Kitodo-API/src/main/java/org/kitodo/serviceloader/KitodoServiceLoader.java]: added

File: Kitodo-API/pom.xml (hunk 2 above)
Comment (henning-gerhardt) [inline Kitodo-API/pom.xml]: Then it should be enough to use artifact with id `log4j-core` or?

Comment (Beacze) [inline Kitodo-API/pom.xml]: changed

**Pull request merged**